Codec/Archive/Tar/Types.hs

```diff
@@ -318,13 +318,12 @@ fromTarPath (TarPath namebs prefixbs) = adjustDirectory $
 -- operating system, eg to perform portability checks.
 --
 fromTarPathToPosixPath :: TarPath -> FilePath
-fromTarPathToPosixPath (TarPath namebs prefixbs) = adjustDirectory $
-  FilePath.Posix.joinPath $ FilePath.Posix.splitDirectories prefix
-                         ++ FilePath.Posix.splitDirectories name
+fromTarPathToPosixPath (TarPath namebs prefixbs) = adjustDirectory $ BS.Char8.unpack $
+   if BS.null prefixbs
+   then namebs
+   else prefixbs <> BS.Char8.pack [FilePath.Posix.pathSeparator] <> namebs
   where
-    name   = BS.Char8.unpack namebs
-    prefix = BS.Char8.unpack prefixbs
-    adjustDirectory | FilePath.Posix.hasTrailingPathSeparator name
+    adjustDirectory | BS.Char8.pack [FilePath.Posix.pathSeparator] `BS.isPrefixOf` namebs
```

Comment:
At the very least `singleton pathSeparator`, but I'd rather avoid `isPrefixOf`, when a simpler function would suffice.

Reply:
Sorry, I don't understand this. The code used to check that `name` has a trailing `/`, but now we check that `namebs` has a leading `/`?.. It does not look right to me at all.

Reply:
🤦

Reply:
What function did you have in mind?